python/ray/air/checkpoint.py

```diff
@@ -524,6 +524,25 @@ def _save_checkpoint_metadata_in_directory(self, path: str) -> None:
         with open(checkpoint_metadata_path, "wb") as file:
             pickle.dump(self._metadata, file)
 
+    def _copy_dir_ignore_conflict(self, src_dir: Path, dst_dir: Path):
+        """This is a workaround for python < 3.8 where shutil.copytree does not
+        support dirs_exist_ok=True.
+
+        We will go through the content of the folder and manually copy ites,
+        while ignoring files that conflict.
+        """
+        for inner in src_dir.iterdir():
+            dest = dst_dir / inner.name
+            if dest.exists():
+                # Ignore files that already exist.
+                # For example, checkpoints from every rank may all have
+                # a same .is_checkpoint file.
+                continue
```

Comment:
This method might over-ignore? If this is my directory:

```
.is_checkpoint
a/
    a.txt
```

And I'm trying to copy over a directory like:

```
.is_checkpoint
a/
    b.txt
```

My `b.txt` wouldn't get copied.

Reply:
Do we have a unit test for this? I think the copy code path is enabled through an env var.

Reply:
ha, got a bit lazy thinking 95% chance we don't have to handle sub-dirs recursively.
you are right though, fixed, and added a unit test.